src/ess/livedata/dashboard/scipp_to_holoviews.py

```diff
@@ -4,20 +4,33 @@
 import scipp as sc
 
 
-def _coord_to_dimension(var: sc.Variable) -> hv.Dimension:
+def coord_to_dimension(var: sc.Variable) -> hv.Dimension:
     """Create a Holoviews Dimension for the coordinate."""
     dim = var.dim
     unit = str(var.unit) if var.unit is not None else None
     return hv.Dimension(dim, label=dim, unit=unit)
 
 
-def _create_value_dimension(data: sc.DataArray) -> hv.Dimension:
+def create_value_dimension(data: sc.DataArray) -> hv.Dimension:
     """Create a Holoviews Dimension for the values."""
     label = data.name if data.name else 'values'
     unit = str(data.unit) if data.unit is not None else None
     return hv.Dimension('values', label=label, unit=unit)
 
 
+def _create_dummy_coord(dim: str, size: int) -> sc.Variable:
+    """Create a dummy coordinate for a missing dimension."""
+    return sc.arange(dim, size, unit=None)
+
+
+def _ensure_coords(da: sc.DataArray) -> sc.DataArray:
+    """Ensure all dimensions have coordinates, creating dummy ones if needed."""
+    for dim in da.dims:
+        if dim not in da.coords:
+            da = da.assign_coords({dim: _create_dummy_coord(dim, da.sizes[dim])})
```

Comment:
Is there an overhead in calling `assign_coords`?
It's probably almost nothing, but if we are calling this several times per second, is it more efficient to collect all the coords in a dict and then assign coords only once at the end?